barretenberg/cpp/src/barretenberg/ultra_honk/decider_proving_key.cpp

```diff
@@ -80,6 +80,110 @@ void DeciderProvingKey_<Flavor>::construct_databus_polynomials(Circuit& circuit)
     }
 }
 
+/**
+ * @brief Check that the number of gates in each block does not exceed its fixed capacity. Move any overflow to the
+ * overflow block.
+ * @details Using a structured trace (fixed capcity for each gate type) optimizes the efficiency of folding. However,
+ * to accommodate circuits which cannot fit into a prescribed trace, gates which overflow their corresponding block are
+ * placed into an overflow block which can contain arbitrary gate types.
+ * @note One sublety is that gates at row i may in general utilize the values at row i+1 via shifts. If the last row in
+ * a full-capacity block is such a gate, then moving the overflow out of sequence will cause that gate not to be
```

Comment:
maybe explain here what you mean by a dummy gate (i.e. that selectors are turn off) and also that it needs duplication so the second to last gate can function appropriately if it needs to check shifts

Reply:
clarified, thanks